tagstudio/src/qt/widgets/thumb_renderer.py

```diff
@@ -759,8 +760,26 @@ def _image_vector_thumb(self, filepath: Path, size: int) -> Image.Image:
             filepath (Path): The path of the file.
             size (tuple[int,int]): The size of the thumbnail.
         """
-        # TODO: Implement.
-        im: Image.Image = None
+
+        # Create an image to draw the svg to and a painter to do the drawing
+        image: QImage = QImage(size, size, QImage.Format.Format_ARGB32)
+        image.fill("#00000000")
+        painter: QPainter = QPainter(image)
+
+        # Create an svg renderer, then render to the painter
+        svg: QSvgRenderer = QSvgRenderer(str(filepath))
+        svg.render(painter)
```

Comment:
how does the thumbnail affect with odd aspect ratios in the SVG?

Reply:
It would have stretched them, but I have now fixed it to keep aspect ratio.